pjsip/src/pjsua-lib/pjsua_media.c

```diff
@@ -2472,9 +2472,6 @@ pj_status_t pjsua_media_channel_init(pjsua_call_id call_id,
 	    call_med->def_dir = PJMEDIA_DIR_ENCODING_DECODING;
 	}
 	call_med->dir = call_med->def_dir;
-	if (call_med->dir == PJMEDIA_DIR_NONE) {
-	    enabled = PJ_FALSE;
-	}
```

Comment:
I suppose this is the modification that fixes the first issue (i.e. the media dir is unaffected)?

If I remember correctly, the reason I included this in the original PR was for the initial offer/answer, to avoid media initialisation if the media direction is set to NONE. So please help test for this case as well to make sure that it works fine, such as:
```
    // video-only call
    call_opt.flag |= PJSUA_CALL_SET_MEDIA_DIR;
    call_opt.media_dir[0] = PJMEDIA_DIR_NONE;
    call_opt.media_dir[1] = PJMEDIA_DIR_DECODING;
```

Reply:
There are two ways of disabling media: by setting port to 0 or by setting direction to inactive, the first will result in a totally disabled media, and the latter the media will still have RTCP traffics (and perhaps RTP keep alive) so media transports are still need to be initialized. As currently we only have one media direction (i.e: PJMEDIA_DIR_NONE), I guess it'd be better to use the latter way?